types/components/buttons.d.ts

```diff
@@ -165,11 +249,26 @@ export interface PayPalButtonsComponentOptions {
     onInit?: (data: Record<string, unknown>, actions: OnInitActions) => void;
     /**
      * Called when the buyer changes their shipping address on PayPal.
+     * @deprecated Use `onShippingAddressChange` or `onShippingOptionsChange` instead.
      */
     onShippingChange?: (
         data: OnShippingChangeData,
         actions: OnShippingChangeActions,
     ) => Promise<void>;
+    /**
+     * Called when the buyer selects a new shipping option on PayPal.
+     */
+    onShippingOptionsChange?: (
+        data: OnShippingOptionsChangeData,
+        actions: OnShippingOptionsChangeActions,
+    ) => Promise<void>;
+    /**
+     * Called when the buyer updates their shipping address on PayPal.
```

Comment:
should this be "changes" instead of "updates"?

Reply:
Oh man, I don't know the difference. A buyer can either add one to begin with or select a different one that's already on file. Does that help? Probably not. 🤣